test/batch_test.cc

```diff
@@ -0,0 +1,39 @@
+#include "benchmark/benchmark.h"
+
+#include <algorithm>
+
+#include "../src/check.h"  // NOTE: check.h is for internal use only!
+
+static void BM_batch_iteration_count(benchmark::State& state) {
+  const size_t batch_size = state.range(0);
+  size_t actual_iterations = 0;
+  while (state.KeepRunningBatch(batch_size)) {
+    actual_iterations += batch_size;
+  }
+  CHECK_GE(state.iterations(), state.max_iterations);
```

Comment:
Sorry I know @dominichamon just suggested the opposite, but I think using `<cassert>` is this test is preferable, just so we can avoid any potential ODR issues for non-debug builds of `libbenchmark.a`.

Ideally the other tests shouldn't use `check.h` either, but the added output turns out to be infinitely helpful in those cases. In this case `<cassert>` should do fine. I'll look into fixing the other tests, or fixing `check.h` to avoid ODR violations in the near future.